src/aiai_eval/evaluator.py

```diff
@@ -281,6 +314,97 @@ def _evaluate_single(
 
+    def _send_results_to_leaderboard(self) -> list:
+        """Send the evaluation results to the leaderboard.
+
+        Args:
+            results (dict):
+                The evaluation results to send.
+
+        Returns:
+            list of bool: A list of booleans indicating whether the results were
+                successfully sent to the leaderboard.
+        """
+        # Initialize a list of status bools
+        status = []
+
+        # Loop through the evaluation results and send each one to the leaderboard
+        for task_name in self.evaluation_results.keys():
+            for model_id in self.evaluation_results[task_name].keys():
+                logger.info(
+                    f"Sending results for {model_id} to the {task_name}-leaderboard."
+                )
+                # Post the results to the leaderboard and log the response
+                try:
+                    task_leaderboard_json = self.leaderboard_client.post_model_to_task(
+                        model_type=self.evaluation_results[task_name][model_id][
+                            "model_type"
+                        ],
+                        task_name=task_name,
+                        model_id=model_id,
+                        metrics=self.evaluation_results[task_name][model_id]["total"],
+                        test=self.evaluation_config.testing,
+                    )
+                    logger.info(
+                        f"Results successfully sent to the {task_name}-leaderboard."
+                    )
+                except (ValueError, ConnectionError) as e:
+                    logger.info(
+                        f"Could not send results for {model_id} to the "
+                        f"{task_name}-leaderboard. Skipping."
+                    )
+                    logger.debug(f'The error message was "{e}".')
+
+                    # Append the status of the leaderboard post to the status list
+                    status.append(False)
+                    continue
+
+                # Make dataframe from leaderboard json
+                task_leaderboard = pd.DataFrame.from_dict(task_leaderboard_json)
+
+                # Log the leaderboard
+                logger.info(
+                    f"Leaderboard:\n{tabulate(task_leaderboard, headers='keys', tablefmt='fancy_grid')}"
+                )
+
+                # Get metric columns, i.e. all columns except "id", "model_type" and "model_id"
+                metric_columns = task_leaderboard.columns.difference(
+                    ["id", "model_type", "model_id"]
+                )
+
+                # Sort the leaderboard by the average of the metric columns
+                task_leaderboard["average_score"] = (
+                    task_leaderboard[metric_columns].dropna(axis=1).mean(axis=1)
+                )
+                task_leaderboard.sort_values("average_score").drop(
+                    "average_score", axis=1
+                )
```

Comment:
Maybe worthwhile to keep the `average_score` column when there are multiple scores?